pkg/store/copr/coprocessor.go

```diff
@@ -1113,20 +1145,62 @@ func (it *copIterator) Next(ctx context.Context) (kv.ResultSubset, error) {
 	return resp, nil
 }
 
+func (it *copIterator) liteSendReq() (resp *copResponse) {
+	defer func() {
+		r := recover()
+		if r != nil {
+			logutil.BgLogger().Error("copIteratorWork meet panic",
+				zap.Any("r", r),
+				zap.Stack("stack trace"))
+			resp = &copResponse{err: util2.GetRecoverError(r)}
+		}
+	}()
+
+	worker := it.liteWorker.worker
+	ctx := it.liteWorker.ctx
+	backoffermap := make(map[uint64]*Backoffer)
+	for len(it.tasks) > 0 {
+		curTask := it.tasks[0]
+		respCh := make(chan *copResponse, 2)
+		curTask.respChan = respCh
+		bo := chooseBackoffer(ctx, backoffermap, curTask, worker)
+		tasks, err := worker.handleTaskOnce(bo, curTask, respCh)
+		if err != nil {
+			resp = &copResponse{err: errors.Trace(err)}
+			worker.checkRespOOM(resp, true)
+			return resp
+		}
+		if len(tasks) > 0 {
+			it.tasks = append(tasks, it.tasks[1:]...)
```

Comment:
Is it possible that there is a remaining task whose batchTaskList is not empty (then multiple copResponses might be pushed into the respCh)?

Reply:
Nice catch, but the `it.liteWorker` only work when there is only 1 task and doesn't have `batchTaskList`, so this case won't happen. https://github.com/pingcap/tidb/pull/57522/files#diff-f341da742ab4e8413960a536dfe0038a2bb92ba2dd6e9f336b5708e19a8b3e60R861

Reply:
But what if the first task meets region error and newly tasks do have `batchTaskList`?

Reply:
Nice catch, already fix it.

Reply:
Is there a test case to cover this situation?